main.cpp

```diff
@@ -5,110 +5,642 @@ Project 5: Part 1 / 4
+}
+
+PlaneWings::~PlaneWings()
+{
+    std::cout << "Jumbo jet wings span " << wingSpan << " feet and have "<< numEngines << " engines attached to them." << std::endl;
+}
+
+void PlaneWings::flex(int flex) 
+{
+    while (flex < maxFlex) 
+    {
+        auto ws = wingSpan;
+        for (ws = 200; ws <= 202; ++ws) 
+        {
+            flex += 1;
+            std::cout << "Longer the wing, more it can flex." << std::endl;
+        }
+    }
+    std::cout << "Wing broken when flexed over 28ft." << std::endl;
+}
+
+void PlaneWings::generateLift (bool upwardForce, std::string airDirection)
+{   
+    std::string planeStatus = "stays on the ground";
+    if (airDirection == "down" && upwardForce)
+    {
+        planeStatus = "goes up";
+    }
+    std::cout << "When the air goes " << airDirection << " the upward force creates lift and the plane " << planeStatus << std::endl;
+}
+
+bool PlaneWings::reduceDrag (float tailwind)
+{
+    if(tailwind < 10.f) 
+    return true;
+    return false;
+}
+void PlaneWings::lowersLandingSpeed (int drag, bool landed, float landingSpeed )
+{
+    if(!landed)
+    {
+    landingSpeed -= drag;
+    }
+}
 
 /*
- new UDT 4:
- with 2 member functions
- */
+new UDT 4:
+with 2 member functions
+*/
+struct Kitchen 
+{
+    Range range;
+    Range::RangeControls rangeControls;
+    
+    Kitchen();
+    ~Kitchen();
+    
+    void designSpace(std::string addSlightVariation, int wallWidth);
+    void safetyAlertViaWiFi(int currentOvenTemp, int setOvenTemp, int currentTime);
+};
+
+Kitchen::Kitchen() 
+{
+    std::cout << "Kitchen being constructed." << std::endl;
+}
+
+Kitchen::~Kitchen() 
+{
+    std::cout << "Kitchen being destructed." << std::endl;
+}
+
+void Kitchen::designSpace(std::string addSlightVariation, int wallWidth) 
+{
+    std::string themeColor =
+    addSlightVariation + rangeControls.controlPanelColor;
+    int measureSpaceForFridge = wallWidth - range.width;
+    
+    std::cout << "Kitchen will be designed to go with " << themeColor
+    << std::endl;
+    std::cout << "We still have enough space for a " << measureSpaceForFridge
+    << " wide fridge." << std::endl;
+}
+
+void Kitchen::safetyAlertViaWiFi(int currentOvenTemp, int setOvenTemp, int currentTime) 
+{
+    if (rangeControls.supportsWifi)
+    {
+        if (currentOvenTemp > 0) 
+        {
+            while (currentTime < 2400 && currentOvenTemp < setOvenTemp) 
+            {
+                currentOvenTemp += 10;
+                int mins;                
+                for (mins = 1; mins <= 10; ++mins) 
+                {
+                    std::cout << "Alert to oven user : oven is on at " << currentOvenTemp << std::endl;
+                }
+            }
+            std::cout << "Auto turn off the oven ." << std::endl;
+        }
+        std::cout << "Oven is not on." << std::endl;
+    }
+    std::cout << "This model doesn't support WiFi feature." << std::endl;
```

Comment:
These two statements are always printed out, regardless of the conditional expressions that come before them.
This seems to go against the behavior you hope to have happen with the `if` statements you're written.
- [ ] revise